src/Component/AutoMapper/Exception/ReadOnlyTargetException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Jane\Component\AutoMapper\Exception;
+
+/**
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+class ReadOnlyTargetException extends RuntimeException
+{
+}
```

Comment:
I would have added a specific message here, for DX: `Cannot use readonly class as an object to populate. You can opt-out this behavior by using the context "context_name"`